src/api/route-services/pipelines.js

```diff
@@ -0,0 +1,196 @@
+const AWS = require('aws-sdk');
+const crypto = require('crypto');
+const fetch = require('node-fetch');
+const YAML = require('yaml');
+const jq = require('jq-web');
+const config = require('../../config');
+
+
+class PipelinesService {
+  constructor() {
+    this.stepFunctions = new AWS.StepFunctions({
+      region: config.awsRegion,
+    });
+
+    this.clusterInfo = null;
+  }
+
+  async getClusterInformation() {
+    if (!this.clusterInfo) {
+      const eks = new AWS.EKS({
+        region: config.awsRegion,
+      });
+
+      const accountId = await config.awsAccountIdPromise();
+      const info = await eks.describeCluster({ name: `biomage-${config.clusterEnv}` }).promise();
+
+      const { name, endpoint } = info.cluster;
+      const certAuthority = info.cluster.certificateAuthority.data;
+      const roleArn = `arn:aws:iam::${accountId}:role/state-machine-role-${config.clusterEnv}`;
+
+      this.clusterInfo = {
+        accountId, name, endpoint, certAuthority, roleArn,
+      };
+    }
+
+    return this.clusterInfo;
+  }
+
+  static async getPipelineImage() {
+    const response = await fetch(
+      config.pipelineInstanceConfigUrl,
+      {
+        method: 'GET',
+      },
+    );
+
+    const txt = await response.text();
+    const manifest = YAML.parseAllDocuments(txt);
+
+    return jq.json(manifest, '..|objects|.r//empty');
```

Comment:
we need handling here on what happens if the GET request returns an error

Reply:
The HTTP server is already rigged up with its own error handler. If there is an error getting the file, it will throw an exception and return with a 500. We can, if you want to, have alternative error handling, but since failing to fetch the config makes it impossible to launch the pipeline, we can fail here hard.

Reply:
as long as we make this code here to return something meaningful if there is an error from the endpoint, I think it should work. The reason why I added comment in here is because it would not be great if it this just blows up silently and then we have to go and figure out what is happening, but maybe this is part of the error handling ticket?

Reply:
Yes, this is part of error handling ticket.

The API always throws an error when things happen on the HTTP side. Those are always caught and an error message is raised on the client, such as the message you see when you can't connect to the API.

Errors are still caught and processed properly on the Socket.IO side as well, it's just that they are not returned back to the UI. For now, this pipeline stuff is an HTTP endpoint because it's easier to test it locally without hooking it up to the UI.

Essentially, the API never blows up under an error condition, it always keeps running, it just silently suppressed exceptions for Socket.IO events, and that is what the ticket to fix that is for.

Reply:
ok, sounds good!